src/App.tsx

```diff
@@ -1,61 +1,134 @@
+
+function getUser(userId: number): User | null {
+  const findUser = usersFromServer.find(user => user.id === userId);
+
+  return findUser || null;
+}
+
+export const todos: Todos[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [newTitle, setNewTitle] = useState('');
+  const [newUser, setNewUser] = useState(todos);
+  const [userIdCount, setUserIdCount] = useState(0);
+
+  const [errorUser, setErrorUser] = useState(false);
+  const [errorTitle, setErrorTitle] = useState(false);
+
+  const changeTitle = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTitle(e.target.value);
+    setErrorTitle(false);
+  };
+
+  const changeUser = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserIdCount(+e.target.value);
+    setErrorUser(false);
+  };
+
+  const hanldeSubmitNewTodo = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    const titleTrim = newTitle.trim();
+
+    setErrorUser(!userIdCount);
+    setErrorTitle(!titleTrim);
+
+    const maxId = Math.max(...newUser.map(todo => todo.id));
+
+    if (titleTrim && userIdCount) {
+      setNewUser(() => [
+        ...newUser,
+        {
+          id: maxId + 1,
+          completed: false,
+          title: titleTrim,
+          userId: userIdCount,
+          user: getUser(userIdCount),
+        },
+      ]);
+
+      setNewTitle('');
+      setUserIdCount(0);
+    }
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={hanldeSubmitNewTodo}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label>
+            Title:
+            <input
+              type="text"
+              data-cy="titleInput"
+              placeholder="Enter a title"
+              value={newTitle}
+              onChange={changeTitle}
+            />
+          </label>
+
+          {errorTitle && (
+            <span className="error">Please enter a title</span>
+          )}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
-          </select>
+          <label>
+            User:
+            <select
+              data-cy="userSelect"
+              value={userIdCount}
+              onChange={changeUser}
+            >
+              <option
+                value="0"
+              >
+                Choose a user
+              </option>
+
+              {usersFromServer.map(user => (
+                <option
+                  value={user.id}
+                  key={user.id}
+                >
+                  {user.name}
+                </option>
```

Comment:
You can use destructuring 
```suggestion
              {usersFromServer.map(({ id, name })=> (
                <option
                  value={id}
                  key={id}
                >
                  {name}
                </option>
```